### DISCUSSION TITLE
Audio working after windows restart

### DISCUSSION THREAD

**Discussion by hla63** [AndresGarciaSobrado91/MSI-Modern15-Hackintosh#4] [Ideas]
Hello ,

I found a way to have audio fully working after windows reboot.
The culprit is the windows driver who not made correctly the audio shutdown.
driver Realtek need to be switch with microsoft driver to properly works after restart
![Capture d’écran 2021-01-02 150146](https://user-images.githubusercontent.com/64965637/103459035-7e5e7d00-4d0c-11eb-8fdd-ea53bcd9c3b5.jpg)

**Comment by hla63** [comment]
I don't know if you also have the wifi disappear sometimes at boot/reboot with the itlwm/airportitlwm.kext but a fix exist
uncheck this case in windows resolve the problem:
![windows fastboot](https://user-images.githubusercontent.com/64965637/103301229-320bf800-4a01-11eb-85bf-fb331f4f9173.png)

**Comment by AndresGarciaSobrado91** [comment]
Hi @hla63 how are you? I hope you are great! I have no other OS installed on my laptop, so, I am not facing this issue. Anyway, this is good information for the future. Thank you very much for sharing this!

**Comment by AndresGarciaSobrado91** [comment]
I have noticed that the webcam stopped working and it is not detected in macOS. So I am trying to solve this. If you have any ideas please let me know.

**Comment by hla63** [comment]
Hi ! Im fine ! Happy new year.
Maybe your webcam is different.Mine is Product 0x211C Vendor 0x5986 SunplusIT

I think your problem is related to USB map.
[Mapping USB Ports in macos](https://github.com/corpnewt/USBMap)

[USBMap.kext.zip](https://github.com/AndresGarciaSobrado91/MSI-Modern15-Hackintosh/files/5777469/USBMap.kext.zip)

Dont forget to uncheck/disable XhciPortLimit

Msi has released new ec firmware update 1551EMS1.106. I've updated without problems.

**Comment by hla63** [comment]
I found a new method to have working sound after restart from windows.
thanks to MikeBeaton & narcyzzo.

I use tool from [Asus website](https://asus-linux.org/blog/sound-2021-01-11/)
[RtHDDump](https://asus-linux.org/RtHDDump_V236.zip)
with these tool I can compare values on Node x20
Microsoft audio Driver versus Realtek audio driver under windows.

[RtHDDump_MICROSOFT driver.txt](https://github.com/AndresGarciaSobrado91/MSI-Modern15-Hackintosh/files/7988578/RtHDDump_MICROSOFT.driver.txt)
[RtHDDump_RTk.txt](https://github.com/AndresGarciaSobrado91/MSI-Modern15-Hackintosh/files/7988579/RtHDDump_RTk.txt)

I found these values diff under Node x20
Index 10 0x0020
Index 1A 0x800b

with [AppleAlc Debug](https://github.com/acidanthera/AppleALC/releases/tag/1.6.8)
There is a tool called alc-verb
Open the terminal go into the appleALC-debug directory
type:
./alc-verb 0x20 0x500 0x10
./alc-verb 0x20 0x400 0x20
./alc-verb 0x20 0x500 0x1a
./alc-verb 0x20 0x480 0x0b

Sound works now !!!

I think these values could be adding to the config data under Pinconfig kext just like [this](https://github.com/acidanthera/AppleALC/pull/758/files)

I need to test & see.

**Reply by thanhhuynhduy** [reply]
I using msi gf63 10sc alc233 and like you when i reboot from windows, macos have no sound, but I'm a newbie, so can you explain some issue:
./alc-verb 0x20 0x400 0x20 -> where do you get "0x400"
./alc-verb 0x20 0x480 0x0b -> where do you get "0x480" and '0x0b"
And how to modify AppleALC by using Pinconfig, thank you.

**Reply by hla63** [reply]
see [here](https://github.com/acidanthera/AppleALC/pull/758) for explanation

0x500 is SET_COEF_INDEX
0x400 is SET_PROC_COEF

0x480 is SET PROC_COEF 0x400+0x80 value & 0b

more explanation [here](https://github.com/acidanthera/bugtracker/issues/1644)

**Reply by hla63** [reply]
For complete procédure ses [here](https://github.com/acidanthera/AppleALC/wiki/Dumping-processing-coefficients)

**Reply by hla63** [reply]
more explanation about the node 20 hidden register
[here](https://docs.kernel.org/sound/hd-audio/realtek-pc-beep.html)

**Comment by hla63** [comment]
It works !!!!!

**Don't forget to add boot-args** alcverbs=1 — to enable alc-verb support (also alc-verbs device property);

My modded [AppleALC.kext.zip](https://github.com/AndresGarciaSobrado91/MSI-Modern15-Hackintosh/files/7988713/AppleALC.kext.zip)
![Capture d’écran 2022-02-02 à 18 13 38](https://user-images.githubusercontent.com/64965637/152204680-ab028906-6a02-4e36-937c-edbe13f03d6e.png)

**Comment by narcyzzo** [comment]
Hi, u do not need alcverbs=1 when u paste verbs to Pin Config Data, u need this boot arg when u using ./alc-verb command to sending verbs, Good Job- glad it works! ;)

**Reply by hla63** [reply]
Right thx.

**Comment by hla63** [comment]
I have pushed [PR layout id 29 ](https://github.com/acidanthera/AppleALC/pull/768) for MSI Modern 15 A10M for AppleALC 1.7.0 I hope ;)

**Comment by schalachi** [comment]
hello I have the same issue. seems to be mostly (but not always) when windows was used. machine is a razer blade base 15 2020. alc ID's (config.plist) I can use include a few like 21, 76,77. if I reboot into macos 11.5.2 a few times it will eventually work. only hardware change is the wifi / Bluetooth card, no bios mods etc can you send me a link to where I can try and resolve this? i can open the apple alc contents in propertree but I really don't understand what to do from there. not computer super but I'm ok with instructions. I did the hackintosh myself using a youtube video (of diff make/model) and modified someone's efi folder and changed by serial lol, excuse if that angry you. if cant help easily no problem! take care guys.

**Reply by hla63** [reply]
try [this](https://github.com/acidanthera/AppleALC/pull/758)
follow the dumping coefficient [procédure](https://github.com/acidanthera/AppleALC/wiki/Dumping-processing-coefficients)
